Packages/com.unity.visualeffectgraph/Editor/Utilities/EventTester/VFXEventTesterWindow.cs

```diff
@@ -38,10 +38,14 @@ public override void OnWillBeDestroyed()
 
             private void OnSelectionChanged()
             {
-                m_Effects = Selection.gameObjects
-                    .Select(x => x.GetComponent<VisualEffect>())
-                    .Where(x => x != null)
-                    .ToArray();
+                List<VisualEffect> tempList = new List<VisualEffect>();
```

Comment:
I'm reluctant to allocate the full size of selected game objects because in large scenes that's many times larger than the number of VisualEffects (selecting the whole scene in Gigaya, for example, is over 30k objects, only 400 of them being VisualEffects). 
I have experimented with pre-allocating with a size of 256 and it's no noticeable improvement in the profile, as the currently remaining ~20ms is nearly entirely spent equally split between getting the Selection.gameobjects and doing the TryGetComponent on each one. This data is only for Gigaya, so obviously might be different in a project where VisualEffects make up a higher proportion of all objects, or a selection of majority VisualEffects instead of the whole scene :)

Reply:
Sounds good! thanks again :)